docs/articles/documentation/guides/basic-guides/best-practices.md

```diff
@@ -0,0 +1,333 @@
+
+* In your test folder, create subfolders for tests that cover different subsystems of your application.
+
+* Don't write long tests. Shorter test scenarios are easier to debug and can run concurrently.
+
+* Any reused data (for example, large sets of reference values or forms inputs) is better stored in a dedicated directory. Consider a descriptive folder name (for instance, `data`).
+
+With all the suggestions applied, your project's file structure might look like this:
+
+```sh
+.
+├── .testcaferc.json
+└── tests
+    ├── |- test_group1/
+    │   └── |-test1.js
+    │       |-test2.js
+    ├── |- test_group2/
+    │   └── |-test1.js
+    │       |-test2.js
+    ├── |- page_model/
+    │   └── |- page1.js
+    │       |- page2.js
+    ├── |- helpers/
+    │   └── |- helper1.js
+    │       |- helper2.js
+    └── |-data
+```
+
+## Setup and Teardown
+
+State management is an integral and important part of web testing. When your tests run, there are inevitably leftovers - database or local storage records, cache, or cookies. This data requires deletion during teardown. Extra steps your app requires to run (like adding records to the database) are performed during setup.
+
+One way to deal with this is to place the setup actions in the preceding test's `after` and `afterEach` hooks. While good for cleanup, these hooks create mutual dependence between your tests when used to set up for a following test. The success of a test is then influenced by a preceding one, which is not desirable.
+
+Consider a test that downloads a file and saves it to a folder. Since the file system persists between tests, this file requires deletion when the test is complete. This is better done with an `afterEach` hook:
+
+```js
+fixture `My fixture`
+    .page `http://example.com`
+    .afterEach( async t => {
+        await cleanDir();
+    });
+
+test('My test', async t => {
+    //test actions
+});
+```
+
+While good for cleanup, `after` and `afterEach` hooks create mutual dependence between tests when used to set up for a following test. The success rate of a test is then influenced by a preceding one, which is not desirable. Such tests need to run in a specific order and can't run in parallel.
+
+Use `before` or `beforeEach` to fullfill your test's prerequisites (for example to create a file necessary for a successful test run):
+
+```js
+fixture `Another fixture`
+    .page `http://example.com`
+    .beforeEach( async t => {
+        await setupFileSystem();
+    });
+
+test('Another test', async t => {
+    //test actions
+});
+```
+
+If the `before`/`beforeEach` setup is unsuccessful, the corresponding test does not run, which saves you time. If `after`/`afterEach` setup yields an error, the following test runs and probably fails due to the unsuccessful setup.
+
+To fulfill your test's prerequisites, use the `before` and `beforeEach` hooks. Clean up with the `after` and `afterEach` hooks.
```

Comment:
`Clean up with the `after` and `afterEach` hooks.`

What are you cleaning up? For example, do you mean "Clean up the tests with..."

Reply:
We clean up the environment the tests run in. When the test finishes, we delete temporary files created during the test run.

Is that ok:

> Clean the environment up with the `after` and `afterEach` hooks.

Reply:
Your suggestion is fine. Please apply this change.

Reply:
Done